docs/tutorials/gluon/learning_rate_finder.md

```diff
@@ -0,0 +1,321 @@
+        min_iter: minimum number of iterations before early stopping can occur (int)
+        """
+        self.smoothing = smoothing
+        self.min_iter = min_iter
+        self.first_loss = None
+        self.running_mean = None
+        self.counter = 0
+        
+    def __call__(self, loss):
+        """
+        loss: from single iteration (float)
+        """
+        self.counter += 1
+        if self.first_loss is None:
+            self.first_loss = loss
+        if self.running_mean is None:
+            self.running_mean = loss
+        else:
+            self.running_mean = ((1 - self.smoothing) * loss) + (self.smoothing * self.running_mean)
+        return (self.running_mean > self.first_loss * 2) and (self.counter >= self.min_iter)
+```
+
+## Usage
+
+Using a Pre-activation ResNet-18 from the Gluon model zoo, we instantiate our Learner and fire up our Learning Rate Finder!
+
+
+```python
+ctx = mx.gpu() if mx.test_utils.list_gpus() else mx.cpu()
+net = mx.gluon.model_zoo.vision.resnet18_v2(classes=10)
+learner = Learner(net=net, data_loader=data_loader, ctx=ctx)
+lr_finder = LRFinder(learner)
+lr_finder.find(lr_start=1e-6)
+```
+
+
+![png](https://raw.githubusercontent.com/dmlc/web-data/master/mxnet/doc/tutorials/lr_finder/finder_plot.png) <!--notebook-skip-line-->
+
+
+As discussed before, we should select a learning rate where the loss is falling (i.e. from 0.001 to 0.05) but before the loss starts to diverge (i.e. 0.1). We prefer higher learning rates where possible, so we select an initial learning rate of 0.05. Just as a test, we will run 500 epochs using this learning rate and evaluate the loss on the final batch. As we're working with a single batch of 128 samples, the variance of the loss estimates will be reasonably high, but it will give us a general idea. We save the initialized parameters for a later comparison with other learning rates.
+
+
+```python
+learner.net.save_params("net.params")
+lr = 0.05
+
+for iter_idx in range(500):
+    learner.iteration(lr=lr)
+    if ((iter_idx % 100) == 0):
+        print("Iteration: {}, Loss: {:.5g}".format(iter_idx, learner.iteration_loss.asscalar()))
+print("Final Loss: {:.5g}".format(learner.iteration_loss.asscalar()))
+```
+
+Iteration: 0, Loss: 2.785 <!--notebook-skip-line-->
+
+Iteration: 100, Loss: 1.6653 <!--notebook-skip-line-->
+
+Iteration: 200, Loss: 1.4891 <!--notebook-skip-line-->
+
+Iteration: 300, Loss: 1.0846 <!--notebook-skip-line-->
+
+Iteration: 400, Loss: 1.0633 <!--notebook-skip-line-->
+
+Final Loss: 1.1812 <!--notebook-skip-line-->
+
+
+We see a sizable drop in the loss from approx. 2.7 to 1.2.
+
+And now we have a baseline, let's see what happens when we train with a learning rate that's higher than advisable at 0.5.
+
+
+```python
+net = mx.gluon.model_zoo.vision.resnet18_v2(classes=10)
+learner = Learner(net=net, data_loader=data_loader, ctx=ctx)
+learner.net.load_params("net.params", ctx=ctx)
```

Comment:
`load_params` is deprecated. Should we use `load_parameters` instead?

Reply:
Changed.